VoxelSniperCore/src/main/java/com/github/kevindagame/snipe/SnipeTool.java

```diff
@@ -29,14 +29,15 @@ protected SnipeTool(@Nullable IBrush brush, SnipeData snipeData) {
         this.snipeData = snipeData;
         messageHelper = new VoxelMessage(snipeData);
         snipeData.setVoxelMessage(messageHelper);
-        if (brush != null && snipeData.owner().getPlayer().hasPermission(brush.getPermissionNode())) { // npe brush.getPermissionNode()
+        if (brush != null && snipeData.owner().getPlayer().hasPermission(brush.getPermissionNode())) {
             this.currentBrush = brush;
         }
     }
 
     public @Nullable IBrush getCurrentBrush() {
         if (currentBrush == null) {
-            return null;
+            currentBrush = snipeData.owner().instantiateBrush(SnipeBrush.class);
```

Comment:
Why has this been added? If brush is null already that probably means that the player doesn't have permission for snipe brush to begin with right?

Reply:
This makes it so that when the player gets the permission after joining, we still set the brush when the player tries to use the brush or a command.

Reply:
Please add this as javadoc or change the method signature. Now the method does something different than the name implies

Reply:
I'll look tomorrow if I could find a better solution for this, otherwise I will make clear what it does.

Reply:
Another option to accomplish this would be to instantiate the default brush in the constructor, even if the player doesn't have the permission. The player still cannot use the brush since we check the permission every time before we do anything.

Reply:
That would work yeah

Reply:
https://github.com/KevinDaGame/VoxelSniper-Reimagined/pull/118/commits/5e33f701a7f19e2be33623fbc499fe22c4ed2597